tools/Dashboard/DevHome.Dashboard/Views/DashboardView.xaml.cs

```diff
@@ -839,8 +863,70 @@ private async void WidgetControl_Drop(object sender, DragEventArgs e)
         }
 
         ViewModel.PinnedWidgets.CollectionChanged += OnPinnedWidgetsCollectionChangedAsync;
+    }
 
-        _log.Debug($"Drop ended");
+    /// <summary>
+    /// Handle keyboard shortcuts for moving widgets left and right.
+    /// </summary>
+    private async void HandleKeyUpAsync(object sender, KeyRoutedEventArgs e)
+    {
+        _log.Debug($"Key up");
+
+        await _moveWidgetsLock.WaitAsync();
+        try
+        {
+            var key = e.Key;
+            _log.Debug($"e.Key = {key}");
+            if (e.Handled || !(key == VirtualKey.Left || key == VirtualKey.Right))
+            {
+                return;
+            }
+
+            var focusedItem = e.OriginalSource as GridViewItem;
+            if (focusedItem == null || focusedItem.Content == null || focusedItem.Content is not WidgetViewModel widgetViewModel)
+            {
+                return;
+            }
+
+            if (IsAltAndShiftPressed())
+            {
+                var index = ViewModel.PinnedWidgets.IndexOf(widgetViewModel);
+                _log.Information($"Move widget {widgetViewModel.WidgetDisplayTitle} at index {index} {key}");
+
+                if (key == VirtualKey.Left && index > 0)

```

Comment:
I believe this can be reduced to figuring out the new index in the if/else block.